src/main/scala/akka/stream/contrib/Sample.scala

```diff
@@ -0,0 +1,74 @@
+package akka.stream.contrib
+
+import java.util.Random
+import java.util.concurrent.ThreadLocalRandom
+
+import akka.stream.stage.{GraphStage, GraphStageLogic, InHandler, OutHandler}
+import akka.stream.{Attributes, FlowShape, Inlet, Outlet}
+
+
+object Sample {
+  /**
+    *
+    * returns every nth elements
+    *
+    * @param nth must > 0
+    * @tparam T
+    * @return
+    */
+  def apply[T](nth: Int): Sample[T] = Sample[T](() => nth)
+
+  /**
+    *
+    * randomly sampling on a stream
+    *
+    * @param maxStep must > 0, default 1000, the randomly step will be between 1 (inclusive) and maxStep (inclusive)
+    * @tparam T
+    * @return
+    */
+  def random[T](maxStep: Int = 1000): Sample[T] = {
+    require(maxStep > 0, "max step for a random sampling must > 0")
+    Sample[T](() => ThreadLocalRandom.current().nextInt(maxStep) + 1)
```

Comment:
Unfortunately the numbers returned by this call are super-bad quality from a scientific point of view. I think this should be detailed in the scaladoc and recommend people that need really good behaved sampling to use their own (XorShift family for example) explicitly.